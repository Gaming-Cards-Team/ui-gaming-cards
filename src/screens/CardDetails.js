import React from "react";
import Header from "../components/Header";
import CardDetailsBody from "../components/CardDetailsBody";
import Footer from "../components/Footer";
import { useParams } from "react-router";

const CardDetails = () => {

  const { id } = useParams();
  return (
    <>
      <Header />
      <CardDetailsBody id={id}/>
      <Footer />
    </>
  );
};

export default CardDetails;
