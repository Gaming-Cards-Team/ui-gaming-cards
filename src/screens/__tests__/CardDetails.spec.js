import React from 'react';
import renderer from 'react-test-renderer';
import CardDetails from '../CardDetails'; 
jest.mock('react-router', () => ({
    useParams: () => ({
      id: '001',
    }),
})); 

describe('Test CardDetails component', () => {

    it('should validate the component', () => {

        const productDetail  = renderer.create(<CardDetails />)
        expect(productDetail).toMatchSnapshot()
    });
});