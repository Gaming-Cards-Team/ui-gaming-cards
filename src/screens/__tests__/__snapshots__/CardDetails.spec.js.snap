// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CardDetails component should validate the component 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-light bg-light fixed-top"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      <img
        alt="Level Up Logo"
        data-testid="logo"
        height="48"
        src="logo.png"
      />
    </a>
  </nav>,
  <div>
    <br />
    <br />
    <br />
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-1 col-sm-12 col-12"
      />
      <div
        className="col-lg-3 col-md-4 col-sm-12 col-12"
        style={
          Object {
            "padding": "40px 0px",
            "textAlign": "center",
          }
        }
      >
        <span
          className=" lazy-load-image-background blur"
          style={
            Object {
              "backgroundImage": "",
              "backgroundSize": "",
              "color": "transparent",
              "display": "inline-block",
              "height": undefined,
              "width": "100%",
            }
          }
        >
          <span
            className=""
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
          />
        </span>
      </div>
      <div
        className="col-lg-4 col-md-4 col-sm-12 col-12"
      >
        <div
          style={
            Object {
              "padding": "40px 20px",
            }
          }
        >
          <div>
            <span
              style={
                Object {
                  "color": "#2d963f",
                  "fontSize": "xx-large",
                  "fontWeight": "bold",
                }
              }
            >
               
            </span>
            <h4 />
            <div
              style={
                Object {
                  "paddingBottom": 15,
                  "paddingTop": 10,
                }
              }
            >
              <div
                className="card"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-6"
                      style={
                        Object {
                          "borderRight": "1px solid #a09797e6",
                        }
                      }
                    >
                      <p>
                        <span
                          style={
                            Object {
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Rarity: 
                        </span>
                      </p>
                      <p>
                        <span
                          style={
                            Object {
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Card#: 
                        </span>
                      </p>
                      <p>
                        <span
                          style={
                            Object {
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Card type: 
                        </span>
                      </p>
                      <p>
                        <span
                          style={
                            Object {
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Affiliation: 
                        </span>
                      </p>
                      <p>
                        <span
                          style={
                            Object {
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Color: 
                        </span>
                      </p>
                      <p>
                        <span
                          style={
                            Object {
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Point value(s): 
                        </span>
                         
                      </p>
                      <p>
                        <span
                          style={
                            Object {
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Uniqueness: 
                        </span>
                      </p>
                    </div>
                    <div
                      className="col-6"
                    >
                      <p
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Description: 
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-lg-3 col-md-3 col-sm-12 col-12"
      />
    </div>
  </div>,
  <nav
    className="navbar navbar-expand navbar-light fixed-bottom"
    style={
      Object {
        "backgroundColor": "#2d963f",
      }
    }
  >
    <div
      data-testid="footer-text"
      style={
        Object {
          "color": "white",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Copyright Star Wars and all associated names and/or images are copyright Lucasfilm Ltd.
    </div>
  </nav>,
]
`;
